src/card/props.js

```
@@ -133,8 +135,24 @@ function validateActionProps (xprops) : ActionProps {
 
   const { action } = xprops
 
-  return {
-    action,
+  switch (action.type) {
+    case "SAVE": {
+      return {
+        ...action,
+        createVaultSetupToken: getCreateVaultSetupToken({
+          createVaultSetupToken: action.createVaultSetupToken,
+        }),
+        onApprove: getSaveActionOnApprove({
+          onApprove: action.onApprove,
+          onError: baseProps.onError,
+        }),
+      };
+    }
+    default: {
```

In card fields i throw an error for the default case here stating that the action isn't supported. Do we want to have consistent behavior here? If so should the behavior be to throw errors for unknown actions or just return it?

---

yeah might as well, since this is exposed to the merchant we could get literally anything here